tests/Doctrine/Tests/DBAL/Functional/NamedParametersTest.php

```diff
@@ -212,6 +212,67 @@ public function testTicket($query,$params,$types,$expected)
             $result[$k] = array_change_key_case($v, CASE_LOWER);
         }
 
-        self::assertEquals($result, $expected);
+        self::assertEquals($expected, $result);
+    }
+
+    public function ticketGitHub3104Provider()
+    {
+        $escapes = [
+            "ESCAPE '\\'",
```

Comment:
So SQLite needs `\\` and all others need `\\\\`…
How to best adjust the test for this?

Reply:
The question is, how to write a portable query which will work on SQLite and others using the DBAL? If there's no proper solution now, then there's nothing to be fixed or tested: no matter how it's implemented, it's not portable.

Reply:
Okay found a way, please have a look again.

And well, the databases require a different query, that is exactly what this PR is fixing. SQLite is broken at the moment, all others work. With this PR SQLite and all others work.

Reply:
@nickvergessen now Postgres is broken. Please take a look at the build.